lib/vector-common/src/internal_event/mod.rs

```diff
@@ -95,8 +95,13 @@ pub fn register<E: RegisterInternalEvent>(event: E) -> E::Handle {
 
 pub type Registered<T> = <T as RegisterInternalEvent>::Handle;
 
+#[derive(Clone, Copy)]
 pub struct ByteSize(pub usize);
 
+#[derive(Clone, Copy)]
+pub struct Count(pub usize);
```

Comment:
Just curious- since the data needs to be cast to `u64` , is it `usize` chosen in the data structure here so that there are fewer overall casts?
Curious because in `ComponentEventsDropped` I used a `u64` with the opposite logic- keeping the type in the data structure consistent with what it will be used as.
Is there is a convention / guideline here?

Reply:
I made the choice to avoid casts in the users of these data types, centralizing the casts into internal events. Any value that is constrained by the memory size (byte sizes, number of elements, etc) will fit into a `usize`, and `usize` will fit into `u64` on all known architectures that we target or plan to target, so it's safe on all counts to just coerce. The reverse is not true if we ever had to coerce away from `u64` to `usize` as that could be 32-bit.

Reply:
FWIW I started down the path of converting all users of this new type to use `usize` in the internal events, but decided that would be out of scope for this work.

Reply:
:+1: Makes sense. Thanks for the context.